src/main/java/com/kodilla/ecommercee/controller/CartController.java

```diff
@@ -1,55 +1,56 @@
 package com.kodilla.ecommercee.controller;
 
-import com.kodilla.ecommercee.domain.CartDto;
-import com.kodilla.ecommercee.domain.OrderDto;
 import com.kodilla.ecommercee.domain.ProductDto;
-import org.springframework.http.MediaType;
+import com.kodilla.ecommercee.mapper.CartMapper;
+import com.kodilla.ecommercee.service.CartDbService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.web.bind.annotation.*;
 
-import java.math.BigDecimal;
-import java.util.ArrayList;
 import java.util.List;
 
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
 @RestController
 @RequestMapping("/v1/carts")
 public class CartController {
-    @PostMapping
-    public void create() {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CartController.class);
+    private CartMapper cartMapper;
+    private CartDbService cartDbService;
+
+    @Autowired
+    public CartController(CartMapper cartMapper, CartDbService cartDbService) {
+        this.cartMapper = cartMapper;
+        this.cartDbService = cartDbService;
     }
 
-    @GetMapping
-    public List<ProductDto> get() {
+    @PostMapping(value="createCart",consumes = APPLICATION_JSON_VALUE)
+    public void createCart(@RequestParam Long userId) {
+        cartDbService.saveCart(cartMapper.mapToCart(cartDbService.createEmptyCart(userId)));
+    }
 
-        List<ProductDto> products = new ArrayList<>();
-        products.add(new ProductDto(1L, "kurtka zimowa", "Pellentesque tempus interdum quam ut rhoncus.", BigDecimal.valueOf(100), 1L));
-        products.add(new ProductDto(2L, "płaszcz", " Vivamus a bibendum purus.", BigDecimal.valueOf(150), 1L));
-        products.add(new ProductDto(8L, "krawat", "Nunc mi mi, laoreet ac mollis nec, pharetra sit amet tortor.", BigDecimal.valueOf(50), 2L));
-        return products;
+    @GetMapping(value="getProductFromCart")
+    public List<ProductDto> getPrductFromCart(@RequestParam Long cartId) {
+        return cartDbService.getProductsFormBasket(cartId);
     }
 
-    @PutMapping(consumes = MediaType.APPLICATION_JSON_VALUE)
-    public List<ProductDto> addProducts(@RequestBody List<ProductDto> products) {
-        products.add(new ProductDto(1L, "kurtka zimowa", "Pellentesque tempus interdum quam ut rhoncus.", BigDecimal.valueOf(100), 1L));
-        products.add(new ProductDto(2L, "płaszcz", " Vivamus a bibendum purus.", BigDecimal.valueOf(150), 1L));
-        products.add(new ProductDto(8L, "krawat", "Nunc mi mi, laoreet ac mollis nec, pharetra sit amet tortor.", BigDecimal.valueOf(50), 2L));
-        products.add(new ProductDto(5L, "torebka", "Nunc mi mi, laoreet ac mollis nec, pharetra sit amet tortor.", BigDecimal.valueOf(40), 3L));
-        return products;
+    @PutMapping(value="addProduct")
+    public void addProduct(@RequestParam Long cartId, @RequestBody ProductDto productDto) {
```

Comment:
```suggestion
    @PutMapping(value="addProduct/{cartId}", consumes = APPLICATION_JSON_VALUE)
    public void addProduct(@PathVariable Long cartId, @RequestBody ProductDto productDto) {
```
Wydaje mi się, że musimy tu dodać consumes, skoro przekazujemy cały obiekt.
Co do parametrów metody umawialiśmy się, że w endpointach stosujemy @PathVariable zamiast @RequestParam.

Reply:
Ok, zmieniłam w kodzie już na @PathVaribale ale przyznaję się że w moim przypadku zdecydowanie wole uzywać @RequestParam, ponieważ są da mnie bardziej czytele szczegółnie wtedy kiedy przekazuję ich kilka parametrów.